src/heitech.zer0mqXt.core/Main/Zer0Mq.cs

```diff
@@ -31,6 +33,18 @@ public IZer0MqBuilder SetLogger(ILogger adapter)
             return this;
         }
 
+        public IZer0MqBuilder SetRetryCount(uint count)
+        {
+            _retryCount = count;
```

Comment:
The user may set `_retryCount` to `0` here, which would result in `SocketConfiguration.RetryCount` having it's value set to `0` after invoking `Build`.
But then, `SocketConfiguration.RetryIsActive` would return `true`, because that `Nullable<uint>` now has a value.
I guess there should be a check somewhere to avoid that -or- having the `SocketConfiguration.RetryIsActive` property not only checking for `SocketConfiguration.RetryCount.HasValue`, but also checking that the `Value != 0`.

Reply:
![image](https://user-images.githubusercontent.com/20025919/154148506-c0eea2b5-5f97-4f94-9852-ee495e608545.png)
the check is here, but the cohesion is hurt by that split up.

So I will change it to a more object oriented approach.

Reply:
done